core/src/test/scala/org/broadinstitute/dsde/rawls/fastpass/FastPassServiceSpec.scala

```diff
@@ -644,4 +689,159 @@ class FastPassServiceSpec
+            WorkbenchEmail(testData.userOwner.userEmail.value + "-pet"),
+            IamMemberTypes.ServiceAccount,
+            IamResourceTypes.Project,
+            workspace.googleProjectId.value,
+            role,
+            expirationDate,
+            DateTime.now()
+          )
+          Seq(userGrant, petGrant)
+      }
+
+      val userBucketGrant = FastPassGrant(
+        0,
+        workspace.workspaceId,
+        WorkbenchUserId(testData.userOwner.userSubjectId.value),
+        WorkbenchEmail(testData.userOwner.userEmail.value),
+        IamMemberTypes.User,
+        IamResourceTypes.Bucket,
+        workspace.bucketName.value,
+        services.terraBucketWriterRole,
+        expirationDate,
+        DateTime.now()
+      )
+
+      val petBucketGrant = FastPassGrant(
+        0,
+        workspace.workspaceId,
+        WorkbenchUserId(testData.userOwner.userSubjectId.value),
+        WorkbenchEmail(testData.userOwner.userEmail.value + "-pet"),
+        IamMemberTypes.ServiceAccount,
+        IamResourceTypes.Bucket,
+        workspace.bucketName.value,
+        services.terraBucketWriterRole,
+        expirationDate,
+        DateTime.now()
+      )
+
+      projectRoles ++ Seq(userBucketGrant, petBucketGrant)
+    }
+
+    allGrants.foreach(grant => runAndWait(fastPassGrantQuery.insert(grant)))
+
+    val startTime = DateTime.now()
+
+    val fastPassGrants1 =
+      runAndWait(fastPassGrantQuery.findFastPassGrantsForWorkspace(testData.workspace.workspaceIdAsUUID))
+    val fastPassGrants2 =
+      runAndWait(fastPassGrantQuery.findFastPassGrantsForWorkspace(testData.workspaceNoAttrs.workspaceIdAsUUID))
+
+    val project1Removal =
+      FastPassService.removeFastPassGrantsInWorkspaceProject(fastPassGrants1,
+                                                             testData.workspace.googleProjectId,
+                                                             slickDataSource.dataAccess,
+                                                             services.googleIamDAO,
+                                                             services.googleStorageDAO,
+                                                             None
+      )(executionContext, services.openTelemetry)
+    val project2Removal =
+      FastPassService.removeFastPassGrantsInWorkspaceProject(fastPassGrants2,
+                                                             testData.workspaceNoAttrs.googleProjectId,
+                                                             slickDataSource.dataAccess,
+                                                             services.googleIamDAO,
+                                                             services.googleStorageDAO,
+                                                             None
+      )(executionContext, services.openTelemetry)
+
+    runAndWait(DBIO.seq(project1Removal, project2Removal))
+
+    val endTime = DateTime.now()
```

Comment:
https://www.baeldung.com/java-8-date-time-intro

Recommend replacing Joda time with with `java.time` libraries.  I think you can replace this stuff with `java.time.Instant` and `java.time.Duration`.

Reply:
No more Joda time in FastPass. Phew, that was quite the brain-scratcher of a change. Time zones are hard!